hilti/runtime/CMakeLists.txt

```diff
@@ -94,6 +93,32 @@ target_compile_definitions(hilti-rt-tests-library-dummy1 PUBLIC RETURN_VALUE=1)
 add_library(hilti-rt-tests-library-dummy2 SHARED src/tests/library-dummy.cc)
 target_compile_definitions(hilti-rt-tests-library-dummy2 PUBLIC RETURN_VALUE=2)
 
+##### Link directories
```

Comment:
I think this might be the first externally visible library we add which is more than a single header. That means the library needs to be able to find its own headers and dependencies under its hardcoded include paths (e.g., `fiber.h` needs to be able to include `<fiber/fiber_mach.h>` or `<hu/lang.h>`). We can adjust paths of includes for our own code, but not this library (w/o forking). For that reason I put it into a "global" location.

An alternative to what I did here would be to add the 3rdparty include directory to the global include directories (e.g., reported by `hilti-config --cxxflags`). That would be messy in a different way. I am not sure there are other options.

Reply:
By following the approach for justrx and by using a pimpl we are able to hide the dependency and I posted patches for that. With my approach `libhilti-rt.a` is still underlinked though:

```.sh
$ nm lib64/libhilti-rt.a | grep fiber_
                 U fiber_init_toplevel
                 U fiber_alloc
                 U fiber_destroy
                 U fiber_reserve_return
                 U fiber_switch
```

I built fiber with a `SHARED` configuration so `-fPIC` gets injected; I then consume the object file like in the case of justrx. With that e.g., `hilti-rt-tests` links and executes successfully since it directly links against the hilti object library, but `libhilti-rt.a` is still underlinked which shows in the btests which cannot find the symbol.

Do you have any idea what I am missing here?

Reply:
Closing this as this should hopefully be addressed by the recently added patches.